### ISSUE TITLE
Looking forward.. (the future of Utf8Json)

### ISSUE THREAD

**Issue by danielmoncada** [neuecc/Utf8Json#170]
@neuecc  Just wanted to check in to see if you still plan on maintaining this project. As of writing, there are currently 22 PRs and 87 issues. Most of these have appeared to not be addressed at all. 

With the release of .NET Core 3, Utf8Json should move forward to support this (i.e., a proper formatter). 

Any updates would be appreciated. Thanks!

**Comment by danielmoncada** [comment]
> I think .NET Core 3 support is essential.
> Also, PR for some bugs will have to be merged.
> 
> MessagePack-CSharp v2 work is nearing completion (this takes more than half a year)
> I think I have time to do them.
> Also, my personal maintenance is completely limited.
> 
> Recruiting maintainers, I will pass a lot of authority.

Thank you for the reply. 

Now that Core 3 is out, there has a been a lot of comparison between Microsoft's System.Text.Json and your Utf8Json. Utf8Json is more than double the speed of Microsoft's. It would be nice to see more maintainer on this so that some of us can incorporate into our pipeline.

Thanks for the time and effort!

**Comment by Svetomechc** [comment]
Yeah, would love to integrate Utf8Json even into my ASP.NET Core 3.0 (and especially my "legacy" 2.2) projects as a default formatter.

**Comment by Svetomechc** [comment]
```cs
.AddMvcOptions(options =>
      {
        options.OutputFormatters.Clear();
        // can pass IJsonFormatterResolver for customize.
        options.OutputFormatters.Add(new JsonOutputFormatter(StandardResolver.CamelCase));
        options.InputFormatters.Clear();
        // if does not pass, library should use JsonSerializer.DefaultResolver.
        options.InputFormatters.Add(new JsonInputFormatter());
      });
```
Okay, this was actually easy, but could've been easier, like:
```cs
.ConfigureUtf8Json()
```
extension method on _IMvcBuilder_.

**Comment by Svetomechc** [comment]
@neuecc I can probably find some time to implement the above, if you are interested.

**Comment by jtreher** [comment]
> > I think .NET Core 3 support is essential.
> > Also, PR for some bugs will have to be merged.
> > MessagePack-CSharp v2 work is nearing completion (this takes more than half a year)
> > I think I have time to do them.
> > Also, my personal maintenance is completely limited.
> > Recruiting maintainers, I will pass a lot of authority.
> 
> Thank you for the reply.
> 
> Now that Core 3 is out, there has a been a lot of comparison between Microsoft's System.Text.Json and your Utf8Json. Utf8Json is more than double the speed of Microsoft's. It would be nice to see more maintainers on this so that some of us can incorporate into our pipeline.
> 
> Thanks for the time and effort!

I enjoy the speed of this as well, but we recently found that when deserializing a large stream, Utf8json used triple the memory and had Gen2 allocations while System.Text.Json was twice as slow, it used 1/3 of the memory and no Gen2 allocations.

**Comment by neuecc** [comment]
MessagePack-CSharp v2 is still rc(but few weeks, will complete).
I've invesitgated new techniques, use IBufferWriter and ReadOnlySequence that exists after Utf8Json relased.
MessagePack v2 takes care of memory usage and Large object heap.
I believe that it will definitely contribute to the performance of Utf8Json.
(However, the implementation of MsgPack v2 takes a year and is not finished yet ...)
